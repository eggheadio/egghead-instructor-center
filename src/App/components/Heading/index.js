import React, {PropTypes} from 'react'

export const levels = ['1', '2', '3', '4', '5']

const Heading = ({children, level}) => {

  const outputByLevel = {
    1: (
      <h1 className='mt0 normal f1 mb4'>
        {children}
      </h1>
    ),
    2: (
      <h2 className='mt0 normal f2 mb3'>
        {children}
      </h2>
    ),
    3: (
      <h3 className='mt0 bold f3 mb2'>
        {children}
      </h3>
    ),
    4: (
      <h4 className='mt0 bold f4 mb2'>
        {children}
      </h4>
    ),
    5: (
      <h5 className='mt0 bold f5 mb2'>
        {children}
      </h5>
    ),
  }

  return (
    <div>
      {outputByLevel[level]}
    </div>
  )
}

Heading.propTypes = {
  children: PropTypes.node.isRequired,
  level: PropTypes.oneOf(levels).isRequired,
}

export default Heading
