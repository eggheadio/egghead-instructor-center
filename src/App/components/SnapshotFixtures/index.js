import React from 'react'

export const StringFixture = 'String'

export const ImageUrlFixture = 'https://s3.amazonaws.com/uifaces/faces/twitter/kimcool/128.jpg'

export const ComponentFixture = () => (
  <div>
    Component
  </div>
)
