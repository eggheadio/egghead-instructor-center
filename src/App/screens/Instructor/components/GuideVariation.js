import React, {PropTypes} from 'react'

const GuideVariation = ({tooltip, children}) => (
  <div title={tooltip}>
    {children}
  </div>
)

GuideVariation.propTypes = {
  tooltip: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default GuideVariation

