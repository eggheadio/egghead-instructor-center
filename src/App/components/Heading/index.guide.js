import React from 'react'
import {storiesOf} from '@kadira/storybook'
import {lorem} from 'faker'
import {map} from 'lodash'
import GuideVariation from '../GuideVariation'
import Heading, {levels} from '.'

storiesOf('Heading', module)

  .add('levels', () => (
    <div>
      {map(levels, level => (
        <GuideVariation tooltip={level}>
          <Heading level={level}>
            {lorem.words()}
          </Heading>
        </GuideVariation>
      ))}
    </div>
  ))
