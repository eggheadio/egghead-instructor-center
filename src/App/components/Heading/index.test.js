import React from 'react'
import {create} from 'react-test-renderer'
import {forEach} from 'lodash'
import {StringFixture} from '../SnapshotFixtures'
import GuideVariation from '../GuideVariation'
import Heading, {levels} from '.'

test('levels', () => {
  forEach(levels, (level) => (
    expect(create(
      <Heading level={level}>
        {StringFixture}
      </Heading>
    )).toMatchSnapshot()
  ))
})
