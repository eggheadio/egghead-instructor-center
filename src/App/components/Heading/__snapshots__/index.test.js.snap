exports[`test levels 1`] = `
<div>
  <h1
    className="mt0 normal f1 mb4">
    String
  </h1>
</div>
`;

exports[`test levels 2`] = `
<div>
  <h2
    className="mt0 normal f2 mb3">
    String
  </h2>
</div>
`;

exports[`test levels 3`] = `
<div>
  <h3
    className="mt0 bold f3 mb2">
    String
  </h3>
</div>
`;

exports[`test levels 4`] = `
<div>
  <h4
    className="mt0 bold f4 mb2">
    String
  </h4>
</div>
`;

exports[`test levels 5`] = `
<div>
  <h5
    className="mt0 bold f5 mb2">
    String
  </h5>
</div>
`;
